infrastructure/smart-contracts/contracts/upgrades/pool-factory/PoolFactory3.sol

```diff
@@ -0,0 +1,875 @@
+// 30: Wait period for this esXai un-stake request is not yet over
+// 31: You must un-stake at least 1 esXai, and the amount must match the un-stake request
+// 32: You must have at least the desired un-stake amount staked in order to un-stake
+// 33: Invalid pool for claim; pool needs to have been created via the PoolFactory
+// 34: Invalid delegate update; pool needs to have been created via the PoolFactory
+// 35: Invalid submission; pool needs to have been created via the PoolFactory
+// 36: Invalid submission; user needs to be owner, delegate or staker to submit
+// 37: Invalid auth: msg.sender must be kyc approved to create a pool
+// 38: Address failed kyc
+
+contract PoolFactory3 is Initializable, AccessControlEnumerableUpgradeable {
+    using EnumerableSetUpgradeable for EnumerableSetUpgradeable.AddressSet;
+
+    // address of the NodeLicense NFT contract
+    address public nodeLicenseAddress;
+
+    // contract addresses for esXai and xai tokens
+    address public esXaiAddress;
+    address public refereeAddress;
+
+    // Boolean to enable/disable staking on the referee contract
+    bool public stakingEnabled;
+
+    // Array to store addresses of staking pool contracts
+    address[] public stakingPools;
+
+    // Maximum share values for staking pool (owner, keys, stakedEsXai) in basepoints (5% => 50,000)
+    uint32[3] public bucketshareMaxValues;
+
+    // Mapping of user addresses to their interacted pool addresses
+    mapping(address => address[]) public interactedPoolsOfUser;
+
+    // Mapping of user addresses to pool addresses to index in user array for efficient removal
+    mapping(address => mapping(address => uint256))
+        public userToInteractedPoolIds;
+
+    // Mapping of delegates to the pools they are delegates of
+    mapping(address => address[]) public poolsOfDelegate;
+
+    // Mapping of pool addresses to indices in the delegate's pools array
+    mapping(address => uint256) public poolsOfDelegateIndices;
+
+    // Mapping of pool addresses created via this factory
+    mapping(address => bool) public poolsCreatedViaFactory;
+
+    // Address of the contract responsible for deploying staking pool and bucket proxies
+    address public deployerAddress;
+
+    // Periods (in seconds) to lock keys/esXai when user creates an unstake request
+    uint256 public unstakeKeysDelayPeriod;
+    uint256 public unstakeGenesisKeyDelayPeriod;
+    uint256 public unstakeEsXaiDelayPeriod;
+
+    // Period (in seconds) to update reward breakdown changes
+    uint256 public updateRewardBreakdownDelayPeriod;
+
+    // Mapping to track if an address failed kyc
+    mapping(address => bool) public failedKyc; // Default is false
+
+    // Role definition for stake keys admin
+    bytes32 public constant STAKE_KEYS_ADMIN_ROLE = keccak256("STAKE_KEYS_ADMIN_ROLE");
+
+    mapping(address => bool) public totalStakeCalculated;
+
+    // =================> VERY IMPORTANT <============================================
```

Comment:
Thats very dramatic :P

Reply:
That should not be our concern right ?
If somebody is building on top of our contracts they will have to read the full code themselves, everything else will be dangerous. What do you think ?

Reply:
I think we may have future devs who join us, may be unfamiliar with the codebase, and think that it's okay to use that variable. 

My concern had nothing to do with someone building on our contracts, my concern was that we don't accumulate more technical debt making the code hard to read and understand.